teleport-server/r53.tf

```diff
@@ -0,0 +1,7 @@
+resource "aws_route53_record" "teleport" {
+  zone_id = "${data.aws_route53_zone.root.zone_id}"
+  name    = "${local.teleport_domain_name}"
+  type    = "CNAME"
+  records = ["${format("ec2-%s.%s.compute.amazonaws.com", replace(aws_eip.teleport_public.public_ip, ".", "-"), data.aws_region.current.name)}"] # Haven't found a better way to get the public_dns of the EIP
```

Comment:
Can't you use this? https://www.terraform.io/docs/providers/aws/r/eip.html#public_ip

Reply:
No, I need the AWS DNS record, that way it'll resolve to the private ip within the VPC (and paired VPCs). If I use the public IP and an A record it'll always resolve to that public ip, even for local connections.

Reply:
So this will not work? https://www.terraform.io/docs/providers/aws/d/instance.html#public_dns

Reply:
After discussing with @iuriaranda this will also not work. Best way is to do it like he did.